### ISSUE TITLE
Apple Books displays question mark (?) instead of images

### ISSUE THREAD

**Issue by hendricius** [hendricius/the-sourdough-framework#59]
For some reason all images except the first one on the cover page are displayed as ? in the book. I re-used the same image. Why wouldn't it display?

The CSS also seems to be the same:
<img width="2413" alt="image" src="https://user-images.githubusercontent.com/816859/225979546-8e48c6aa-c50a-48e0-bca3-31ba0377915b.png">

Another issue on SO mentions it could be the file size (https://stackoverflow.com/questions/6342605/ibooks-my-epub-doesnt-show-png-images-a-question-mark-is-in-its-place-instead/75770633#75770633). But then I used the same image, made it even smaller. I tried portrait or landscape mode. The image still only displays as `?`. Upon clicking you can see the full version of the image.

Maybe a book in iBooks?

**Comment by hendricius** [comment]
Latest findings: https://www.loom.com/share/389a00fbc0b54fcfa121f51f098fd883

**Comment by ceigel** [comment]
According to this https://blog.kotobee.com/optimize-images-ebook/ the image should be 300-400 pixels wide

**Comment by hendricius** [comment]
@ceigel this won't fix it unfortunately. I think it's related to the https://github.com/michal-h21/tex4ebook plugin that I use to generate the pdf.

The strange thing: If I put the images into a separate latex document and include them, that works very well. This is the same thing I am doing with all the flow charts and tables. However, when added directly to the main document, they don't show. I investigated the HTML too and it looks the same. I am quite clueless.

**Comment by mart-e** [comment]
![IMG_20230522_143400](https://github.com/hendricius/the-sourdough-framework/assets/564822/e1f00dbf-4133-4b83-bc0f-e146e71953a8)

This is how images are (not) rendered on my Kobo using the epub. I guess it's the same root cause.

I am sure this was on purpose to make me buy the physical version of the book ! :money_with_wings:

**Comment by hendricius** [comment]
> ![IMG_20230522_143400](https://user-images.githubusercontent.com/564822/239910505-e1f00dbf-4133-4b83-bc0f-e146e71953a8.jpg)
> 
> This is how images are (not) rendered on my Kobo using the epub. I guess it's the same root cause. (If not sorry for the noise, I will move to a separate issue)
> 
> I am sure this was on purpose to make me buy the physical version of the book ! 💸

What about the flow charts and tables? Are they properly displayed?

**Comment by mart-e** [comment]
> What about the flow charts and tables? Are they properly displayed?

They are not but nor are they on my PC book reader:
[Flow chart kobo](https://github.com/hendricius/the-sourdough-framework/assets/564822/ae325891-abf3-4ddd-afba-5f0ae1573159)
[Flow chart PC](https://github.com/hendricius/the-sourdough-framework/assets/564822/143cb6de-c4a3-4b9e-99bd-ed7151880016)

[Table kobo](https://github.com/hendricius/the-sourdough-framework/assets/564822/c811525d-b88e-4129-8e62-fd46ef1e2114)
[Table PC](https://github.com/hendricius/the-sourdough-framework/assets/564822/a04ed863-89a5-449e-8943-d72a3e865b55)

**Comment by hendricius** [comment]
Thanks for the pictures. So when integrating pictures like I do with the tables/figures (https://github.com/hendricius/the-sourdough-framework/blob/main/book/wheat-sourdough/wheat-sourdough.tex#L45) they seem to show up properly. In that case I included another LaTeX document. When adding them directly, they don't show up.

The problem that they are not properly sized in your case seems to be another problem. https://github.com/michal-h21/tex4ebook/issues/32. If you modify the tex4ebook.cfg and change to:

```
\Preamble{xhtml}
\begin{document}
\Configure{DocumentLanguage}{en}
\Configure{OpfScheme}{URI}
\Configure{UniqueIdentifier}{https://the-bread-code.io}
\Configure{CoverMimeType}{image/jpeg}
\CoverMetadata{images/cover-page.jpg}
\Css{img{max-width: 100%;}}
\EndPreamble
```

Note the `max-width` I added. Does it change it? Just run `make bake` to build the PDF versions again :-)

Does it improve the result?

**Comment by hendricius** [comment]
Also small update - I had recently made this PR to test some changes: https://github.com/hendricius/the-sourdough-framework/pull/60/files

**Comment by hendricius** [comment]
Some more screenshots attached. In Apple Books the embedded figures are displayed properly:

Works:
```
\begin{figure}[!htb]
\begin{center}
  \includegraphics{tables/table-bakers-math-example.pdf}
  \caption{An example table demonstrating how to properly calculate using baker's math}
\end{center}
\end{figure}
```

Does not work:

```
\begin{figure}[!htb]
  \includegraphics[width=\textwidth]{sourdough-starter.jpg}
  \caption{A very active sourdough starter shown by the bubbles in the dough}
  \label{fig:sourdough-starter}
\end{figure}
```

Then I built the ebooks using `make build_ebook`

I then used the Calibre Viewer to inspect the `.epub`. It seems to ship with chrome so that I could use the inspector tools and see. The resulting HTML seems to be very similar. I couldn't notice a difference really except maybe the max-width parameter (which I then also stripped).

A solution could be to include all images the same way like I include the tables and flowcharts. But this would be quite some work to move them all out of each .tex file.

Screenshots:
<img width="2503" alt="html-latex-embedded-document" src="https://github.com/hendricius/the-sourdough-framework/assets/816859/9f66ce92-05ae-49a2-a6a3-5ae91ef4e0e7">
<img width="1699" alt="html-embedded-image" src="https://github.com/hendricius/the-sourdough-framework/assets/816859/32a4511a-971a-4d61-9548-32050083ab07">
<img width="963" alt="apple-books-not-working" src="https://github.com/hendricius/the-sourdough-framework/assets/816859/880d3b7b-5b77-453c-9bb9-cff6fbe94833">
<img width="1295" alt="works-calibre" src="https://github.com/hendricius/the-sourdough-framework/assets/816859/450106e2-d851-46e1-ac0b-8712f8859450">
